### DISCUSSION TITLE
Separate `docker` directory into a separate repository

### DISCUSSION THREAD

**Discussion by developomp** [supabase/supabase#11801] [Feature Requests]
# Chore

## Describe the chore

Create a new repository containing what's currently inside the [`docker`](https://github.com/supabase/supabase/tree/master/docker) directory.

## Additional context

Currently, a clean clone of the https://github.com/supabase/supabase repository weighs over 2GB in size even with the `--depth 1` flag. This is way more than acceptable when we want to automate the deployment of supabase instances.

GitHub workflow such as https://github.com/BetaHuhn/repo-file-sync-action can be used to leave the directory as it is while making it available in another repo.

**Comment by kiwicopple** [comment]
Ah yeah - that's a fair point. We're looking at moving the images/videos out of the main repo (to Supabase Storage), so I think that will reduce the size of the repo. Let us do that first, and if then we can revisit this one.

My reluctance to move it out of the main repo right now is maintainence - by putting it here, we can pay more attention to the github issues that are raised (like your one) so that self-hosting isn't neglected. As you can see in some of our other repos, they aren't as active as this one.

I'll convert this one to a Discussion for now. Thanks @developomp 👍

**Comment by ruggi99** [comment]
Hi @developomp 
PR #11871 should shrink down context and improve building with docker. Have a look at it if you want.

**Reply by developomp** [reply]
Thanks for sharing but my issue is about the repository size. Not the docker image size.

**Reply by ruggi99** [reply]
I thought you were talking about the context sent to docker daemon during docker build but I completely misunderstood the whole point. Sorry about that
